priceupdater/priceupdater.go

```diff
@@ -208,14 +286,123 @@ func (p *PriceUpdater) UpdateTokenList() error {
 	// For each token from the DB
 	for _, dbToken := range dbTokens {
 		// If the token doesn't exists in the config list,
-		// add it with default update emthod
-		if _, ok := p.tokensConfig[dbToken.Addr]; !ok {
-			p.tokensConfig[dbToken.Addr] = TokenConfig{
-				UpdateMethod: p.defaultUpdateMethod,
-				Symbol:       dbToken.Symbol,
-				Addr:         dbToken.Addr,
+		// add it with default update method
+		if _, ok := p.statictokensMap.Statictokens[dbToken.TokenID]; ok {
+			continue
+		} else {
+			if !(p.providers[0].SymbolsMap.Symbols[dbToken.TokenID] == UpdateMethodTypeIgnore ||
+				p.providers[0].AddressesMap.Addresses[dbToken.TokenID] == common.FFAddr) {
+				p.tokensList[dbToken.TokenID] = dbToken
+			}
+		}
+		for i := 0; i < len(p.providers); i++ {
+			if len(p.providers[i].SymbolsMap.Symbols) != 0 {
+				if _, ok := p.providers[i].SymbolsMap.Symbols[dbToken.TokenID]; !ok {
+					p.providers[i].SymbolsMap.Symbols[dbToken.TokenID] = dbToken.Symbol
+				}
+			}
+			if len(p.providers[i].AddressesMap.Addresses) != 0 {
+				if _, ok := p.providers[i].AddressesMap.Addresses[dbToken.TokenID]; !ok {
+					p.providers[i].AddressesMap.Addresses[dbToken.TokenID] = dbToken.Addr
+				}
 			}
 		}
 	}
 	return nil
 }
+
+type fiatExchangeAPI struct {
+	Base  string
+	Rates interface{}
+}
+
+func (p *PriceUpdater) getFiatPrices(ctx context.Context) (map[string]interface{}, error) {
+	var url = "latest?base=" + p.fiat.BaseCurrency + "&symbols=" + p.fiat.Currencies + "&access_key=" + p.fiat.APIKey
+	req, err := p.clientProviders["fiat"].New().Get(url).Request()
+	if err != nil {
+		return make(map[string]interface{}), tracerr.Wrap(err)
+	}
+	var res *http.Response
+	var result map[string]interface{}
+	var data *fiatExchangeAPI
+	res, err = p.clientProviders["fiat"].Do(req.WithContext(ctx), &data, nil)
+	if err != nil {
+		return make(map[string]interface{}), tracerr.Wrap(err)
+	}
+	if data != nil {
+		result = data.Rates.(map[string]interface{})
+	} else {
+		log.Error("Error: data got are empty. Http code: ", res.StatusCode, ". URL: ", url)
+	}
+	return result, nil
+}
+
+// UpdateFiatPrices updates the fiat prices
+func (p *PriceUpdater) UpdateFiatPrices(ctx context.Context) error {
+	log.Debug("Updating fiat prices")
+	//Retrieve fiat prices
+	prices, err := p.getFiatPrices(ctx)
+	if err != nil {
+		return tracerr.Wrap(err)
+	}
+	//Getting all price from database with baseCurrency USD
+	currencies, err := p.db.GetAllFiatPrice("USD")
+	if err != nil {
+		return tracerr.Wrap(err)
+	}
+	for token, pr := range prices {
+		price := pr.(float64)
+		var exist bool
+		for i := 0; i < len(currencies); i++ {
+			if token == currencies[i].Currency {
+				exist = true
+			}
+		}
+		if exist {
+			if err = p.db.UpdateFiatPrice(token, "USD", price); err != nil {
+				log.Error("DB error updating fiat currency price: ", token, ", ", price, " Error: ", err)
```

Comment:
Why there error is only printed and not returned?

Reply:
It is returned but after process all the currency prices